apps/provider.c

```diff
@@ -165,8 +165,7 @@ int provider_main(int argc, char **argv)
     prog = opt_init(argc, argv, provider_options);
     while ((o = opt_next()) != OPT_EOF) {
         switch (o) {
-        case OPT_EOF:
-        case OPT_ERR:
+        default: /* Catching OPT_ERR & covering OPT_EOF which isn't possible */
```

Comment:
?????  How come coverity hasn't caught the *exact same thing* in `apps/engine.c`

Also, a reason we don't have `default:` in any of our sub-commands is to catch programming errors, such as forgetting to treat an option.  So using it here is the wrong thing to do.

So maybe there's something else that we can do (such as figuring out why coverity doesn't complain about the other sub-commands...  or simply treat it as a false positive.

Reply:
My guess would be that the other errors were already tagged as intentional or false positives.  This is existing code that has moved to a new file, it should have been clean.  BTW: I've seen Coverity complain about similar relocations before.

Is `default:` that unreasonable in the _error_ branch of the `switch`?  Forgetting to treat an option and returning error seems reasonable.  Something might sneak through if the PR author isn't checking the code they wrote and didn't write a unit test.  I'd hope the reviewers would catch the latter.

If it is that important to not use `default:`, I'd suggest returning an `enum` instead of an `int` from `opt_next` to limit the possible range of return values.

Reply:
> Is `default:` that unreasonable in the _error_ branch of the `switch`?  Forgetting to treat an option and returning error seems reasonable.  Something might sneak through if the PR author isn't checking the code they wrote and didn't write a unit test.  I'd hope the reviewers would catch the latter.

The original intention was that the omission would be detected by the compiler.  However:

> If it is that important to not use `default:`, I'd suggest returning an `enum` instead of an `int` from `opt_next` to limit the possible range of return values.

... meh, so moot point :confused:

Reply:
Which way to swing here?

Reply:
[shrug] I'm dropping this concern.